web/src/settings_org.ts

```diff
@@ -1249,6 +1250,33 @@ export let init_dropdown_widgets = (): void => {
         return options;
     };
 
+    const private_notification_channel_options = (): dropdown_widget.Option[] => {
+        const streams = stream_settings_data.get_streams_for_settings_page();
+        const options: dropdown_widget.Option[] = streams
+            .filter((stream) => stream.invite_only)
+            .map((stream) => ({
+                name: stream.name,
+                unique_id: stream.stream_id,
+                stream,
+            }));
+
+        const disabled_option = {
+            is_setting_disabled: true,
+            show_disabled_icon: true,
+            show_disabled_option_name: false,
+            unique_id: DISABLED_STATE_ID,
+            name: $t({defaultMessage: "Disabled"}),
+        };
```

Comment:
Let's move this option outisde to be commonly used for `notification_stream_options` and `private_notification_channel_options`.